### PR TITLE
Prevent bees loading chunks checking hive position

### LINKED ISSUE #2820: Bee movement - server soft locks, all players lose connection indefinitely.
I have LOTS of this in my logs.  It was so prevalent that I rolled the server back to 1.14.4 by restoring a backup.  Every instance has bee movement in it, this is just one of many examples.  In case it's important...  one of our worlds has no floor - think sky islands.

```
DO NOT REPORT THIS TO PAPER - THIS IS NOT A BUG OR A CRASH  - git-Paper-31 (MC: 1.15.1)
The server has not responded for 180 seconds! Creating thread dump
------------------------------
Server thread dump (Look for plugins here before reporting to Paper!):
The server is waiting on these chunks: [[( -1299,-990) in 'world']]
------------------------------
Current Thread: Server thread
	PID: 25 | Suspended: false | Native: false | State: TIMED_WAITING
	Stack:
		sun.misc.Unsafe.park(Native Method)
		java.util.concurrent.locks.LockSupport.parkNanos(LockSupport.java:215)
		net.minecraft.server.v1_15_R1.IAsyncTaskHandler.bk(IAsyncTaskHandler.java:131)
		net.minecraft.server.v1_15_R1.IAsyncTaskHandler.awaitTasks(IAsyncTaskHandler.java:120)
		net.minecraft.server.v1_15_R1.ChunkProviderServer.getChunkAt(ChunkProviderServer.java:285)
		net.minecraft.server.v1_15_R1.World.getChunkAt(World.java:330)
		net.minecraft.server.v1_15_R1.IWorldReader.getChunkAt(IWorldReader.java:87)
		net.minecraft.server.v1_15_R1.World.getChunkAt(World.java:274)
		net.minecraft.server.v1_15_R1.World.getChunkAtWorldCoords(World.java:269)
		net.minecraft.server.v1_15_R1.World.getTileEntity(World.java:1069)
		net.minecraft.server.v1_15_R1.WorldServer.getTileEntity(WorldServer.java:214)
		net.minecraft.server.v1_15_R1.World.getTileEntity(World.java:1043)
		net.minecraft.server.v1_15_R1.EntityBee.eP(EntityBee.java:389)
		net.minecraft.server.v1_15_R1.EntityBee.movementTick(EntityBee.java:378)
		net.minecraft.server.v1_15_R1.EntityLiving.tick(EntityLiving.java:2381)
		net.minecraft.server.v1_15_R1.EntityInsentient.tick(EntityInsentient.java:275)
		net.minecraft.server.v1_15_R1.EntityBee.tick(EntityBee.java:169)
		net.minecraft.server.v1_15_R1.WorldServer.entityJoinedWorld(WorldServer.java:722)
		net.minecraft.server.v1_15_R1.WorldServer$$Lambda$3534/1764556432.accept(Unknown Source)
		net.minecraft.server.v1_15_R1.World.a(World.java:878)
		net.minecraft.server.v1_15_R1.WorldServer.doTick(WorldServer.java:488)
		net.minecraft.server.v1_15_R1.MinecraftServer.b(MinecraftServer.java:1239)
		net.minecraft.server.v1_15_R1.DedicatedServer.b(DedicatedServer.java:417)
		net.minecraft.server.v1_15_R1.MinecraftServer.a(MinecraftServer.java:1106)
		net.minecraft.server.v1_15_R1.MinecraftServer.run(MinecraftServer.java:933)
		java.lang.Thread.run(Thread.java:748)
------------------------------
--- DO NOT REPORT THIS TO PAPER - THIS IS NOT A BUG OR A CRASH ---
```

### PR DESCRIPTION
**Pull request by BillyGalbreath**
Fixes #2820

### CODE REVIEW

File: Spigot-Server-Patches/0420-Prevent-bees-loading-chunks-checking-hive-position.patch [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+diff --git a/src/main/java/net/minecraft/server/EntityBee.java b/src/main/java/net/minecraft/server/EntityBee.java
+index fb1c0a96..8b5d34b2 100644
+--- a/src/main/java/net/minecraft/server/EntityBee.java
++++ b/src/main/java/net/minecraft/server/EntityBee.java
+@@ -386,6 +386,7 @@ public class EntityBee extends EntityAnimal implements EntityBird {
+         if (!this.hasHivePos()) {
+             return false;
+         } else {
++            if (world.getChunkIfLoadedImmediately(hivePos.x >> 4, hivePos.z >> 4) == null) return true; // Paper - just assume the hive is still there, no need to load the chunk(s)
```
Comment (Spottedleaf) [inline Spigot-Server-Patches/0420-Prevent-bees-loading-chunks-checking-hive-position.patch]: getX, getZ

**Review by Spottedleaf — APPROVED** [review]

**Review by zachbr — APPROVED** [review]

**Comment by Ghost-chu** [comment]
Solutions to the symptoms and not the root causes.
Paper should optimize the chunk loading speed/override the mojang sucks code not just say hey, fuck chunk loading, disallow.

**Comment by BillyGalbreath** [comment]
> Solutions to the symptoms and not the root causes.
> Paper should optimize the chunk loading speed/override the mojang sucks code not just say hey, fuck chunk loading, disallow.

PRs welcome ^_^

**Comment by electronicboy** [comment]
I've stated this a dozen times over and will state it once again, loading specifically is *not* the issue with 1.14+ especially with papers optimizations here, chunk access, in general, is slower and there is work pending to try to improve that even further; preventing loading is a key step to improving server performance, however; Loading involves steps which are slow and cannot be avoided, e.g. disk access

**Pull request merged**